components/aws/sagemaker/common/_utils.py

```diff
@@ -581,6 +603,211 @@ def get_best_training_job_and_hyperparameters(client, hpo_job_name):
     return best_job, train_hyperparameters
 
 
+def create_workteam(client, args):
+    ### Documentation: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html#SageMaker.Client.create_workteam
+    """Create a workteam"""
+    with open('/app/common/workteam.template.yaml', 'r') as f:
+        request = yaml.safe_load(f)
+
+    request['WorkteamName'] = args['team_name']
+    request['Description'] = args['description']
+
+    if args['sns_topic']:
+    	request['NotificationConfiguration']['NotificationTopicArn'] = args['sns_topic']
+    else:
+        request.pop('NotificationConfiguration')
+
+    for group in [n.strip() for n in args['user_groups'].split(',')]:
+        request['MemberDefinitions'].append({'CognitoMemberDefinition': {'UserPool': args['user_pool'], 'UserGroup': group, 'ClientId': args['client_id']}})
+
+    for key, val in args['tags'].items():
+        request['Tags'].append({'Key': key, 'Value': val})
+
+    try:
+        response = client.create_workteam(**request)
+        portal = client.describe_workteam(WorkteamName=args['team_name'])['Workteam']['SubDomain']
+        logging.info("Labeling portal: " + portal)
+        return response['WorkteamArn']
+    except ClientError as e:
+        raise Exception(e.response['Error']['Message'])
+
+
+def create_labeling_job_request(args):
+    ### Documentation: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html#SageMaker.Client.create_labeling_job
+    with open('/app/common/gt.template.yaml', 'r') as f:
+        request = yaml.safe_load(f)
+
+    # Mapping are extracted from ARNs listed in https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html#SageMaker.Client.create_labeling_job
+    algorithm_arn_map = {'us-west-2': '081040173940',
+              'us-east-1': '432418664414',
+              'us-east-2': '266458841044',
+              'eu-west-1': '568282634449',
+              'ap-northeast-1': '477331159723',
+              'ap-southeast-1': '454466003867'}
+
+    task_map = {'bounding box': 'BoundingBox',
+              'image classification': 'ImageMultiClass',
+              'semantic segmentation': 'SemanticSegmentation',
+              'text classification': 'TextMultiClass'}
+
+    auto_labeling_map = {'bounding box': 'object-detection',
+              'image classification': 'image-classification',
+              'text classification': 'text-classification'}
```

Comment:
what if I want to stage a custom image labeling task? I don't see that case here?

Reply:
in terms of automatic data labeling for custom labeling jobs - Ground Truth currently doesn't support that, which is why this map only contains the built-in algos that have auto data labeling support
in general though, custom tasks are handled while updating pre-human and annotation consolidation task lambda functions